ktor-server/ktor-server-config-yaml/jvmAndNix/src/io/ktor/server/config/yaml/YamlConfig.kt

```diff
@@ -43,15 +55,29 @@ public class YamlConfig(private val yaml: YamlMap) : ApplicationConfig {
         val parts = path.split('.')
         val yaml = parts.dropLast(1).fold(yaml) { yaml, part -> yaml[part] as? YamlMap ?: return null }
         val value = yaml[parts.last()] ?: return null
-        return ConfigValue(value, path)
+        return when (value) {
+            is YamlLiteral -> resolveValue(value.content, root)?.let { LiteralConfigValue(key = path, value = it) }
+
+            is YamlList -> {
+                val values = value.content.map { element ->
+                    element.asLiteralOrNull()?.content?.let { resolveValue(it, root) }
```

Comment:
Can't it be a nested list of lists?

Reply:
we support only lists of strings and lists of objects, which are accessed through `configList(path)` method